packages/components/src/unit-control/index.tsx

```diff
@@ -168,6 +171,50 @@ function UnforwardedUnitControl(
 				refInputSuffix.current?.focus();
 			}
 		};
+		onPaste = ( event: ClipboardEvent< HTMLInputElement > ) => {
```

Comment:
It might be worth extracting this as a helper function with dependency injection so it's easier to unit test the logic.